excursionset_rundown.md

```diff
@@ -0,0 +1,16 @@
+
+#21cmFAST in MP-Gadget rundown
+
+*Syncpoints are set up every 10 Myr or so (free parameter) in order to trigger the excursion set.
+*Whenever a star particle is created, mass is added to a spatial grid where it formed.
+*When we reach one of the UV syncpoints, stellar mass and total mass grids are created by nearest gridpoint, and decomposed into a slab for each task.
+*We then step through filters of different scales using FFTW and count photons in each cell (from stellar mass * photons per baryon) to determine which cells are ionised, and calculate J21 if so.
+*This J21 is then converted into ionisation and photoheating rates at a certain spectral slope (free parameter) and passed into the existing cooling functions.
+
+
+##Notes:
+
+*Conversion factors from J21 to rates are passed in a table file via the same parameter as the TREECOOL files, if the excursion set flag is on or off it will treat this file as a conversion table or TREECOOL file repsectively.
```

Comment:
Could you create a new option for the excursion set file? It could be ignored if the excursion set flag is off.

Reply:
Do you mean just have another string option which is passed into init_cooling instead when the excursion set is on? Or do you mean completely separate the loading and storage of each table, so they could possibly both be loaded in the same run?

Reply:
I think my answer depends on another question: how does the excursion set model connect to the normal cooling code? I think that at the moment once a cell is reionized, J21 is set to zero and it just uses the rates as the old cooling code doe. But the excursion set code is still called every few Myr? Is this because the tail of reionization may finish very (arbitrarily) late?

Reply:
J21 is zero until the cell is ionized. After this, the J21 calculated by the excursion set (via the SFR) is multiplied by the photorates given in this table at the chosen spectral slope, which is passed into the cooling code instead of the TREECOOL redshift-dependent photo-rates. As the code exists right now, only one of these files will be used.

Reply:
I see, I misread the conditions, sorry. 

I think my concern is really that post-reionization we should retain the good agreement with the lyman alpha forest - I'm worried that once quasars take over the SFR will no longer be a good evolution proportionality constant for the UVB and we need a redshift dependent TREECOOL as well. But I might easily be misunderstanding the model, so please correct me if I'm wrong.

Reply:
PS I'm happy to merge this as a reionization era model and not worry about that: people who want lyman-alpha can continue with the older model. But to enable it for the big run we would need to make sure it played nicely with post-reionization physics.

Reply:
That's a good point, I'll separate the loading for now, and later on put in a way to switch back to the global UVBG after reionisation